fsthttp/imageopto/imageopto.go

```diff
@@ -0,0 +1,787 @@
+
+const (
+	MetadataCopyright Metadata = "copyright"
+)
+
+func (m Metadata) isSet() bool    { return m != "" }
+func (m Metadata) String() string { return string(m) }
+
+type Profile string
+
+const (
+	ProfileBaseline Profile = "baseline"
+	ProfileMain     Profile = "main"
+	ProfileHigh     Profile = "high"
+)
+
+func (p Profile) isSet() bool    { return p != "" }
+func (p Profile) String() string { return string(p) }
+
+type ResizeAlgorithm string
+
+const (
+	ResizeAlgorithmNearest  ResizeAlgorithm = "nearest"
+	ResizeAlgorithmBilinear ResizeAlgorithm = "bilinear"
+	ResizeAlgorithmBicubic  ResizeAlgorithm = "bicubic"
+	ResizeAlgorithmLanczos2 ResizeAlgorithm = "lanczos2"
+	ResizeAlgorithmLanczos3 ResizeAlgorithm = "lanczos3"
+)
+
+func (r ResizeAlgorithm) isSet() bool    { return r != "" }
+func (r ResizeAlgorithm) String() string { return string(r) }
+
+type Sharpen struct {
+	Amount    uint8
+	Radius    float32
+	Threshold uint8
+}
+
+func (s *Sharpen) String() string {
+	return fmt.Sprintf("a%v,r%v,t%v", s.Amount, fmtFloat(float64(s.Radius)), s.Threshold)
+}
+
+func (s *Sharpen) validate() error {
+	if s.Amount > 10 {
+		return errors.New("imageopto: sharpen amount out of range 0 .. 10")
+	}
+	if s.Radius < 0.5 || s.Radius > 1000 {
+		return errors.New("imageopto: sharpen radius out of range 0.5 .. 1000")
+	}
+
+	return nil
+}
+
+type EnableOpt string
+
+const (
+	EnableOptUpscale EnableOpt = "upscale"
+)
+
+func (e EnableOpt) isSet() bool    { return e != "" }
+func (e EnableOpt) String() string { return string(e) }
+
+type Opts struct {
+	Region                             Region
+	PreserveQueryStringOnOriginRequest bool
+	Auto                               Auto
+	BgColor                            *HexColor
+	Blur                               BlurMode
+	Brightness                         int
+	Bw                                 BWMode
+	Canvas                             *Canvas
+	Contrast                           int
+	Crop                               *Crop
+	Dpr                                float32
+	Enable                             EnableOpt
+	Fit                                Fit
+	Format                             Format
+	Frame                              uint32
+	Height                             PixelsOrPercentage
+	Level                              Level
+	Metadata                           Metadata
+	Optimize                           OptimizeLevel
+	Orient                             Orientation
+	Pad                                *Sides
+	Precrop                            *Crop
+	Profile                            Profile
+	Quality                            uint32
+	ResizeFilter                       ResizeAlgorithm
+	Saturation                         int
+	Sharpen                            *Sharpen
+	Trim                               *Sides
+	TrimColor                          *TrimColor
+	Width                              PixelsOrPercentage
+}
+
+func encodeCommas(s string) string {
```

Comment:
```suggestion
func encodeCommasAndColons(s string) string {
```

Reply:
Removed in the end.  I only added this so I could use the Rust test cases unmodified, but imageopto doesn't need them to be encoded.